sdk/remoterendering/mixedreality-remoterendering/src/internal/expectDefinedFromService.ts

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * Check that obj[k] is defined. If not, throw a specific error.
+ * @internal
+ */
+export function expectDefinedFromService<T, K extends keyof T>(obj: T, k: K): NonNullable<T[K]> {
```

Comment:
this is pretty crazy to me, why are we doing error checking on the server response shape? Surely you don't expect clients to try to be resilient against the server returning incomplete responses?  Also what would a user do with the thrown error?

Reply:
The constraints implied here are not imposed by the REST API. Responses from the service could _in theory_ set or not set any of the properties. In reality, there is structure to how the service sets the property, but this exists only as a convention. It's therefore possible that a service change might unintentionally violate the convention. While we would hopefully catch that change elsewhere before deployment, this code would at least trigger a sensible failure in the nightly SDK pipelines.

The user can't do anything useful in this situation (hence "Internal Error"): The service would be at fault. I believe that an Internal Error is still better than unexpected behavior, which might cause the user to try to add unnecessary workarounds.

I'm willing to hear alternative suggestions.

Reply:
If you want to assert something from the server in nightlies, I think the only real solution is to write a live test that asserts on the response payload. Putting these checks in here (and throwing an unrecoverable error to the consumer) feels like a waste of CPU cycles.